# Introduction
For this project, you will interpret data from the National Parks Service **about endangered species** in different parks.

You will perform some data analysis on the **conservation statuses of these species and investigate if there are any patterns** or themes to the types of species that become endangered. During this project, you will analyze, clean up, and plot data as well as pose questions and seek to answer them in a meaningful way.

After you perform your analysis, you will share your findings about the National Park Service.

Questions that this project has sought to answer:

- What is the distribution of conservation status for species?
- Are certain types of species more likely to be endangered?
- Are the differences between species and their conservation status significant?
- Which animal is most prevalent and what is their distribution amongst parks?

**Data sources:**

Both `Observations.csv` and `Species_info.csv` was provided by [Codecademy.com](https://www.codecademy.com).

Note: The data for this project is *inspired* by real data, but is mostly fictional.

### Analysis

In this section, descriptive statistics and data visualization techniques will be employed to understand the data better. Statistical inference will also be used to test if the observed values are statistically significant. Some of the key metrics that will be computed include: 

1. Distributions
1. counts
1. relationship between species
1. conservation status of species
1. observations of species in parks.

## Import modules

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

## Loading data

In [4]:
species = pd.read_csv("species_info.csv")
observations = pd.read_csv("observations.csv")

### Data exploration of `Species.csv`

In [5]:
print("This df has a shape of:", species.shape)

This df has a shape of: (5824, 4)


In [44]:
species.describe()

category    scientific_name        common_names  \
count             5824               5824                5824   
unique               7               5541                5504   
top     Vascular Plant  Castor canadensis  Brachythecium Moss   
freq              4470                  3                   7   

       conservation_status  
count                  191  
unique                   4  
top     Species of Concern  
freq                   161

In [16]:
species.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5824 entries, 0 to 5823
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   category             5824 non-null   object
 1   scientific_name      5824 non-null   object
 2   common_names         5824 non-null   object
 3   conservation_status  191 non-null    object
dtypes: object(4)
memory usage: 182.1+ KB


In [17]:
for col in species.columns:
    print(f"{col} values: {species[col].unique()}")
    print()

category values: ['Mammal' 'Bird' 'Reptile' 'Amphibian' 'Fish' 'Vascular Plant'
 'Nonvascular Plant']

scientific_name values: ['Clethrionomys gapperi gapperi' 'Bos bison' 'Bos taurus' ...
 'Parthenocissus vitacea' 'Vitis californica' 'Tribulus terrestris']

common_names values: ["Gapper's Red-Backed Vole" 'American Bison, Bison'
 'Aurochs, Aurochs, Domestic Cattle (Feral), Domesticated Cattle' ...
 'Thicket Creeper, Virginia Creeper, Woodbine'
 'California Grape, California Wild Grape'
 'Bullhead, Caltrop, Goathead, Mexican Sandbur, Puncture Vine, Texas Sandbur']

conservation_status values: [nan 'Species of Concern' 'Endangered' 'Threatened' 'In Recovery']



In [6]:
print(f"Number of species:{species.scientific_name.nunique()}")

Number of species:5541


In [7]:
species.groupby("category").size()

category
Amphibian              80
Bird                  521
Fish                  127
Mammal                214
Nonvascular Plant     333
Reptile                79
Vascular Plant       4470
dtype: int64

Vacular Plants is the biggest category by far

In [8]:
species.groupby("conservation_status").size()

conservation_status
Endangered             16
In Recovery             4
Species of Concern    161
Threatened             10
dtype: int64

In [9]:
print( "Without conservation status", round(species["conservation_status"].isna().sum()/ len(species["conservation_status"]),6))
print( "Without conservation status", round((len(species["conservation_status"]) - species["conservation_status"].isna().sum())/ len(species["conservation_status"]),6))

Without conservation status 0.967205
Without conservation status 0.032795


Most species are not in concern (5633). But there are 191 species under a conservation status, most of them are species of concern

The `species_info.csv` contains information on the different species in the National Parks (5824 entries). The columns in the data set include:
- **category** - The category of taxonomy for each species
- **scientific_name** - The scientific name of each species
- **common_names** - The common names of each species
- **conservation_status** - The species conservation status

It seems that most of the values in **conservation_status** are *NaN*

There are a 7 categories Mammal, Bird, Reptile, Amphibian, Fish, Vascular Plant and Nonvascular Plant

There are 5541 species, most of them belongs to the category of Vascular plants

### Data exploration of `observations.csv`

In [10]:
observations.head()

scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85

In [17]:
print("This df shapes is of:", observations.shape)

This df shapes is of: (23296, 3)


In [11]:
observations.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23296 entries, 0 to 23295
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   scientific_name  23296 non-null  object
 1   park_name        23296 non-null  object
 2   observations     23296 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 546.1+ KB


There aren't any NaN values in this df

In [12]:
observations["observations"].describe()

count    23296.000000
mean       142.287904
std         69.890532
min          9.000000
25%         86.000000
50%        124.000000
75%        195.000000
max        321.000000
Name: observations, dtype: float64

In [20]:
print("Total observations:", observations.observations.sum())

Total observations: 3314739


In [19]:
print("Number of unique species observed:", observations["scientific_name"].nunique())

Number of unique species observed: 5541


`observation.csv` have the same amount of species as `species_info.csv`

In [21]:
print(f"Number of parks:{observations.park_name.nunique()}")
print(f"parks names:{observations.park_name.unique()}")

Number of parks:4
parks names:['Great Smoky Mountains National Park' 'Yosemite National Park'
 'Bryce National Park' 'Yellowstone National Park']
